vllm/core/scheduler.py

```diff
@@ -396,6 +404,12 @@ def schedule(self) -> Tuple[List[SequenceGroupMetadata], SchedulerOutputs]:
                 if scheduler_outputs.prompt_run else None,
             )
             seq_group_metadata_list.append(seq_group_metadata)
+
+        # Now that the batch has been created, we can assume all blocks in the
+        # batch will have been computed before the next scheduling invocation.
+        for seq_group in scheduler_outputs.scheduled_seq_groups:
```

Comment:
This assumes the model execution won't fail & there's no retry. It is the general assumption in the engine now?

Reply:
yep

Reply:
comment added

> What happen to later preemption

these lines do not change prefix caching behavior wrt later preemption (either case requires recomputing the blocks)

Reply:
oh btw @rkooo567 this assumption is not introduced in this PR; the scheduler already assumes that its schedule is implemented by the engine. retries and failures are not yet in scope of vllm